pylsp/plugins/flake8_lint.py

```diff
@@ -20,6 +20,29 @@
     "F523",  # .format(...) unused positional arguments
     "F841",  # local variable `name` is assigned to but never used
 }
+ERROR_CODES = {
+    # Errors from flake8 itself
+    "E999",  # syntax error
+    # Errors from the pyflakes plugin of flake8
+    #
+    # The following list should be kept in sync with the `pyflakes_lint` plugin.
+    # Here is a snippet to generate such a list:
+    #
+    #     from flake8.plugins.pyflakes import FLAKE8_PYFLAKES_CODES
+    #     from pylsp.plugins.pyflakes_lint import PYFLAKES_ERROR_MESSAGES
+    #     for m in PYFLAKES_ERROR_MESSAGES:
+    #         print(f'"{FLAKE8_PYFLAKES_CODES[m.__name__]}",  # {m.__name__}')
```

Comment:
I suggest to use something similar to this code to keep Pyflakes and Flake8 error codes in sync without manual maintenance.

In other words, I suggest to build `ERROR_CODES` like this (with imports in the appropriate places, of course):

```python
from flake8.plugins.pyflakes import FLAKE8_PYFLAKES_CODES

from pylsp.plugins.pyflakes_lint import PYFLAKES_ERROR_MESSAGES

ERROR_CODES = {FLAKE8_PYFLAKES_CODES[m.__name__] for m in PYFLAKES_ERROR_MESSAGES} | {"E999"}
```

Reply:
Thanks for the review!

I was considering doing so but decided to maintain it manually because:

  - Flake8 does [not guarantee a stable API](https://github.com/PyCQA/flake8/blob/018b0c7f2a525d13f80f5fa3e196f98de4bab284/docs/source/user/python-api.rst), so `FLAKE8_PYFLAKES_CODES` is somewhat internal.
  - We run `flake8` as an executable and parse its outputs, so using the `flake8` module may bring import errors or inconsistencies (e.g., the user installs Flake8 in a different virtual environment from pylsp).

Honestly both are quite subtle though. If you think they are not problems I will change to what you suggest. Thank you!